packages/plugin-table/src/TableCellNode.ts

```diff
@@ -0,0 +1,159 @@
+}
+
+export class TableCellNode extends VNode {
+    attributes: TableCellAttributes;
+    // Only the `managerCell` setter should modify the following private keys.
+    __managerCell: TableCellNode;
+    __managedCells = new Set<TableCellNode>();
+
+    constructor(public header = false) {
+        super();
+    }
+
+    //--------------------------------------------------------------------------
+    // Lifecycle
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return a new VNode with the same type and attributes as this VNode.
+     *
+     *  @override
+     */
+    clone(): this {
+        const clone = new this.constructor<typeof TableCellNode>(this.header);
+        clone.attributes = { ...this.attributes };
+        return clone;
+    }
+
+    //--------------------------------------------------------------------------
+    // Getters
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    get name(): string {
+        let coordinatesRepr = ' <(' + this.rowIndex + ', ' + this.columnIndex + ')';
+        if (this.colspan > 1 || this.rowspan > 1) {
+            const endRow = this.rowIndex + this.rowspan - 1;
+            const endColumn = this.columnIndex + this.colspan - 1;
+            coordinatesRepr += ':(' + endRow + ', ' + endColumn + ')';
+        }
+        coordinatesRepr += '>';
+        return (
+            super.name +
+            coordinatesRepr +
+            (this.header ? ': header' : '') +
+            (this.isActive ? '' : ' (inactive)')
+        );
+    }
+    /**
+     * Return the cell that manages this cell, if any.
+     */
+    get managerCell(): TableCellNode {
+        return this.__managerCell;
+    }
+    /**
+     * Set the given cell as manager of this cell. If null is passed, this cell
+     * will have no manager.
+     */
+    set managerCell(newManager: TableCellNode | null) {
+        const oldManager = this.__managerCell;
+        this.__managerCell = newManager;
+
+        if (newManager) {
+            // If we just set a new manager for this cell, also add this cell to
+            // the new manager's managed cells.
+            newManager.__managedCells.add(this);
+
+            // Copy the manager.
+            this.attributes = { ...newManager.attributes };
+            this.header = newManager.header;
+
+            // Move the children to the manager.
+            newManager.append(...this.children);
+
+            // Hand the managed cells over to the manager.
+            for (const managedCell of this.managedCells) {
+                managedCell.managerCell = newManager;
+                this.__managedCells.delete(managedCell);
+            }
+        } else if (oldManager) {
+            // If we just removed this cell's manager, also remove this cell
+            // from the old manager's managed cells.
+            oldManager.__managedCells.delete(this);
```

Comment:
Why is this in a `else if` ? I mean, you had an `oldManager`, I'm assignining you a `newManager`, don't you have to update `oldManager` anyway ?